src/main.c

```diff
@@ -59,22 +61,35 @@ static void usage(int code) {
     "For more advanced filters see the jq(1) manpage (\"man jq\")\n"
     "and/or https://stedolan.github.io/jq\n\n"
     "Example:\n\n\t$ echo '{\"foo\": 0}' | jq .\n"
-    "\t{\n\t\t\"foo\": 0\n\t}\n\n"
-    "Some of the options include:\n"
-    "  -c               compact instead of pretty-printed output;\n"
-    "  -n               use `null` as the single input value;\n"
-    "  -e               set the exit status code based on the output;\n"
-    "  -s               read (slurp) all inputs into an array; apply filter to it;\n"
-    "  -r               output raw strings, not JSON texts;\n"
-    "  -R               read raw strings, not JSON texts;\n"
-    "  -C               colorize JSON;\n"
-    "  -M               monochrome (don't colorize JSON);\n"
-    "  -S               sort keys of objects on output;\n"
-    "  --tab            use tabs for indentation;\n"
-    "  --arg a v        set variable $a to value <v>;\n"
-    "  --argjson a v    set variable $a to JSON value <v>;\n"
-    "  --slurpfile a f  set variable $a to an array of JSON texts read from <f>;\n"
-    "\nSee the manpage for more options.\n", JQ_VERSION, progname);
+    "\t{\n\t\t\"foo\": 0\n\t}\n\n",
+    JQ_VERSION, progname, progname, progname);
+  if (keep_it_short) {
+    fprintf(f,
+      "For a listing of options, use %s --help.\n",
+      progname);
+  } else {
+    (void) fprintf(f,
```

Comment:
Oh, that's just habit from my days working on Solaris, where the C-style guidelines required casting ignored return values to (void).  IIRC some lints actually do the right thing with them  (Whereas GCC will not let you quiet warnings about ignoring important return values this way.  Always a reason to spit curses its way.)